pkg/services/ngalert/state/historian/annotation.go

```diff
@@ -156,12 +161,12 @@ func (h *AnnotationBackend) buildAnnotations(rule *ngmodels.AlertRule, states []
 	return items
 }
 
-func (h *AnnotationBackend) recordAnnotationsSync(ctx context.Context, panel *panelKey, annotations []annotations.Item, logger log.Logger) {
+func (h *AnnotationBackend) recordAnnotationsSync(ctx context.Context, panel *panelKey, annotations []annotations.Item, logger log.Logger) error {
 	if panel != nil {
 		dashID, err := h.dashboards.getID(ctx, panel.orgID, panel.dashUID)
 		if err != nil {
 			logger.Error("Error getting dashboard for alert annotation", "dashboardUID", panel.dashUID, "error", err)
-			return
+			return fmt.Errorf("Error getting dashboard for alert annotation: %w", err)
```

Comment:
I wonder if we still need logging here. Probably log on the receiving side?

Reply:
Only benefit I can think of is extra context fields, like dashboardUID it was trying to lookup